### PR TITLE
Python: Fix IndexError in Azure OpenAI stream requests by handling empty choices array

### LINKED ISSUE #3650: Python: Stream request errors with Azure OpenAI API version 2023-06-01-preview and later
**Describe the bug**
After specifying Azure OpenAI API version 2023-06-01-preview or later for stream requests, an error occurs due to an unexpected empty choices array in the stream response. This issue is specific to Azure OpenAI and relates to the inclusion of responsible AI content filtering results in responses from this version onward.

**To Reproduce**

```py
from semantic_kernel.connectors.ai.chat_request_settings import ChatRequestSettings
from semantic_kernel.connectors.ai.open_ai import AzureChatCompletion


chat_service = AzureChatCompletion(
    api_version='2023-09-01-preview',
    endpoint='your-endpoint',
    api_key='your-api-key',
    deployment_name='your-deployment-name',
)
stream = chat_service.complete_chat_stream_async(
    [{'role': 'user', 'content': 'Hello, how are you?'}],
    ChatRequestSettings(),
)

async for data in stream:
    print(data)
```

This leads to the following exception

```py
    302 def _parse_choices(chunk):
    303     message = ""
--> 304     if "role" in chunk.choices[0].delta:
    305         message += chunk.choices[0].delta.role + ": "
    306     if "content" in chunk.choices[0].delta:

IndexError: list index out of range
```

**Expected behavior**
When sending stream requests with Azure OpenAI, it is expected that no IndexError occurs and the response is successfully retrieved.

**Screenshots**
N/A

**Platform**
N/A

**Additional context**
This issue appears to stem from a change in the API response format introduced with the 2023-06-01-preview version of Azure OpenAI. This version includes responsible AI content filtering results in the responses, leading to an initial empty choices array in the stream response, which contradicts the implicit assumption of the AzureTextCompletion and AzureChatCompletion stream requests that choices length is always one or more. The sample code provided above can be used to replicate the error. The relevant API change is documented [here](https://github.com/Azure/azure-rest-api-specs/pull/25880).

Here is an example of a stream response.

```json
{"id":"","choices":[],"created":0,"model":"","object":"","system_fingerprint":null,"prompt_filter_results":[{"prompt_index":0,"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}}}]}
{"id":"xxx","choices":[{"delta":{"content":null,"function_call":null,"role":"assistant","tool_calls":null},"finish_reason":null,"index":0,"content_filter_results":{}}],"created":1700739226,"model":"gpt-35-turbo","object":"chat.completion.chunk","system_fingerprint":"yyy"}
{"id":"xxx","choices":[{"delta":{"content":"1","function_call":null,"role":null,"tool_calls":null},"finish_reason":null,"index":0,"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}}}],"created":1700739226,"model":"gpt-35-turbo","object":"chat.completion.chunk","system_fingerprint":"yyy"}
...
{"id":"xxx","choices":[{"delta":{"content":null,"function_call":null,"role":null,"tool_calls":null},"finish_reason":"stop","index":0,"content_filter_results":{}}],"created":1700739226,"model":"gpt-35-turbo","object":"chat.completion.chunk","system_fingerprint":"yyy"}
```

### PR DESCRIPTION
**Pull request by KanchiShimono**
### Motivation and Context

Closes #3650 

Fixing errors in stream requests with Azure OpenAI API version 2023-06-01-preview and later.

### Description

- Added handling for cases where the chunk size in stream responses is zero in the Azure OpenAI Service class.
- Modified integration tests to cover multiple versions of the Azure OpenAI API.

Added two variables to GitHub Actions for integration testing: `AZUREOPENAI__APIVERSIONS` and `AZUREOPENAI__CHAT__APIVERSIONS`. To ensure successful integration tests, these variables need to be added to the repository.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by KanchiShimono** [comment]
@microsoft-github-policy-service agree

File: python/tests/integration/completions/test_azure_oai_chat_service.py [hunk 1]
```diff
@@ -8,9 +8,15 @@
 import semantic_kernel.connectors.ai.open_ai as sk_oai
 
 
+def get_aoai_chat_api_versions():
```
[the remark this answers is not in the document]
Comment (KanchiShimono) [reply python/tests/integration/completions/test_azure_oai_chat_service.py]: @moonbox3 
Thank you for your reply.
I agree with your opinion. To enhance reusability, I have moved the function to `test_utils.py`.

**Comment by moonbox3** [comment]
Hi @KanchiShimono, we just upgraded the OpenAI SDK to be 1.0 or higher (#3555). As part of this work (a large PR), we made some [improvements](https://github.com/microsoft/semantic-kernel/blob/1d51d5d1eb93f40fe888676c101d8cd5799b22f7/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion_base.py#L48C15-L48C36) around handling responses that may contain empty content, similar to what you show here in your PR. Can you grab latest code and please verify that it works for you -- and if not, you can use this PR to update whatever needs a fix. Thank you!

**Comment by KanchiShimono** [comment]
The issue has been resolved.
Thank you!

**Pull request closed without merging**